mm/iob/iob_initialize.c

```diff
@@ -30,14 +30,36 @@
 
 #include "iob.h"
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define ROUNDUP(x, y)     (((x) + (y) - 1) / (y) * (y))
+
+/* Fix the I/O Buffer size with specified alignment size */
+
+#define IOB_ALIGN_SIZE    ROUNDUP(sizeof(struct iob_s), CONFIG_IOB_ALIGNMENT)
+#define IOB_BUFFER_SIZE   (IOB_ALIGN_SIZE * CONFIG_IOB_NBUFFERS + \
+                           CONFIG_IOB_ALIGNMENT - 1)
+
 /****************************************************************************
  * Private Data
  ****************************************************************************/
 
-/* This is a pool of pre-allocated I/O buffers */
+/* Following raw buffer will be divided into iob_s instances, the initial
+ * procedure will ensure that the member io_head of each iob_s is aligned
+ * to the CONFIG_IOB_ALIGNMENT memory boundary.
+ */
+
+static uint8_t g_iob_buffer[IOB_BUFFER_SIZE]
+#ifdef CONFIG_IOB_SECTION
```

Comment:
Would you object to a #define IOB_LOCATION the is defined in Pre-processor Definitions 
```
#ifdef CONFIG_IOB_SECTION
#  define   IOB_LOCATION locate_data(CONFIG_IOB_SECTION)
#else
#  define   IOB_LOCATTION 
#endif
```
and used here?

```static uint8_t g_iob_buffer[IOB_BUFFER_SIZE] IOB_LOCATION;```

Reply:
how about change IOB_LOCATION to IOB_SECTION?

Reply:
Sure